src/libcore/ptr.rs

```diff
@@ -2509,6 +2509,36 @@ pub fn eq<T: ?Sized>(a: *const T, b: *const T) -> bool {
     a == b
 }
 
+/// Hash the raw pointer address behind a reference, rather than the value
+/// it points to.
+///
+/// # Examples
+///
+/// ```
```

Comment:
You'll need to add `#![feature(ptr_hash]` at the top; it won't work otherwise. :)